lib/HLSL/HLModule.cpp

```diff
@@ -1085,7 +1085,7 @@ void HLModule::MergeGepUse(Value *V) {
       }
     }
   }
-  if (V->user_empty()) {
+  if (V->user_empty() && bDelete) {
```

Comment:
Can you just rely on dead code to delete the unused value? Do you have to explicitly delete it here?